packages/amplify-cli/src/domain/amplify-toolkit.ts

```diff
@@ -442,4 +444,20 @@ export class AmplifyToolkit {
       require(path.join(this._amplifyHelpersDirPath, 'get-imported-auth-properties')).getImportedAuthProperties;
     return this._getImportedAuthProperties;
   }
+
+  constructor() {
+    this._cleanUpTasks = new Array();
+  }
+
+  addCleanUpTask = (task: Function) => {
+    this._cleanUpTasks.push(task);
+  }
+
+  getCleanUpTasks = (): Function[] => {
+    return this._cleanUpTasks;
+  }
+
+  runCleanUpTasks = async (context: Context) => {
+    await Promise.all(this._cleanUpTasks.map(task => task(context)));
```

Comment:
I'm still pushing for passing `this` into the task and removing it as a parameter from `runCleanUpTasks` but I won't insist on it ;)

Reply:
discussed offline, it's good as is